packages/back-end/src/events/handlers/utils.ts

```diff
@@ -0,0 +1,182 @@
+          : [],
+      };
+
+    case "feature.updated":
+      return {
+        tags: uniq(
+          (event.data.current.tags || []).concat(event.data.previous.tags || [])
+        ),
+        projects: uniq(
+          (event.data.current.project
+            ? [event.data.current.project]
+            : []
+          ).concat(
+            event.data.previous.project ? [event.data.previous.project] : []
+          )
+        ),
+      };
+
+    case "feature.deleted":
+      return {
+        tags: event.data.previous.tags || [],
+        projects: event.data.previous.project
+          ? [event.data.previous.project]
+          : [],
+      };
+
+    case "experiment.created":
+      return {
+        tags: event.data.current.tags || [],
+        projects: event.data.current.project
+          ? [event.data.current.project]
+          : [],
+      };
+
+    case "experiment.updated":
+      return {
+        tags: uniq(
+          (event.data.current.tags || []).concat(event.data.previous.tags || [])
+        ),
+        projects: uniq(
+          (event.data.current.project
+            ? [event.data.current.project]
+            : []
+          ).concat(
+            event.data.previous.project ? [event.data.previous.project] : []
+          )
+        ),
+      };
+
+    case "experiment.deleted":
+      return {
+        tags: event.data.previous.tags || [],
+        projects: event.data.previous.project
+          ? [event.data.previous.project]
+          : [],
+      };
+
+    default:
+      invalidEvent = event;
+      throw `Invalid event: ${invalidEvent}`;
+  }
+};
+
+export const filterEventForEnvironments = ({
+  event,
+  environments,
+}: {
+  event: NotificationEvent;
+  environments: string[];
+}): boolean => {
+  let invalidEvent: never;
+
+  switch (event.event) {
+    case "user.login":
+      return false;
+
+    case "experiment.created":
+    case "experiment.updated":
+    case "experiment.deleted":
+      return true;
+
+    case "feature.created":
+    case "feature.deleted":
+      return true;
+
+    case "feature.updated":
+      return filterFeatureUpdatedNotificationEventForEnvironments({
+        featureEvent: event,
+        environments,
+      });
+
+    default:
+      invalidEvent = event;
+      throw `Invalid event: ${invalidEvent}`;
+  }
+};
+
+const filterFeatureUpdatedNotificationEventForEnvironments = ({
+  featureEvent,
+  environments,
+}: {
+  featureEvent: FeatureUpdatedNotificationEvent;
+  environments: string[];
+}): boolean => {
+  const { previous, current } = featureEvent.data;
+
+  if (previous.archived && current.archived) {
+    // Do not notify for archived features
+    return false;
+  }
+
+  // Manual environment filtering
+  const changedEnvironments = new Set<string>();
+
+  // Some of the feature keys that change affect all enabled environments
+  const relevantKeysForAllEnvs: (keyof ApiFeature)[] = [
+    "archived",
+    "defaultValue",
+    "project",
+    "valueType",
```

Comment:
we probably need to add checks for "prerequisites". Changes here should definitely trigger an update event.